### PR TITLE
add epbf program to trace podman cleanup errors in CI

### PR DESCRIPTION
**Pull request by Luap99**
See commits, I just want to check if this even works. Hopefully this allows me to properly understand cleanup errors.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Luap99** [comment]
In case anyone is interested here is an example log:
https://api.cirrus-ci.com/v1/artifact/task/6197821693493248/cleanup_tracer/podman-cleanup-tracer.log

It doesn't look nice mostly because the argv lines is printed with zero bytes not spaces, I don't have a way to get rid of that in my program but I guess I could pipe into sed to replace all zero bytes with spaces to make it easier to read.

**Comment by Luap99** [comment]
@edsantiago please pick this one as well for the no retry PR, maybe I can find the cause in this new log.

**Comment by edsantiago** [comment]
Here ya go [f39 root](https://api.cirrus-ci.com/v1/artifact/task/5831003032256512/html/int-podman-fedora-39-root-host-boltdb.log.html#t--Podman-kube-play-multi-doc-yaml-with-multiple-services--pods-and-deployments--1). This is my second run; first one did not have any relevant flakes. This run had only one. I'll rerun at least once more today.

**Comment by edsantiago** [comment]
This one has two ([debian root](https://api.cirrus-ci.com/v1/artifact/task/5511752060764160/html/int-podman-debian-13-root-host-sqlite.log.html#t--Podman-kube-play-test-restartPolicy--1)). It was the only failure in the third run. I've pushed one more time, but will not be checking its status again tonight.

**Comment by Luap99** [comment]
Ok that doesn't seems to include any new things AFAICS, it does confirm what I have been saying the cleanup is called twice. The only way I see this happening is what the cleanup process failed but in the logs exit codes I clearly see no errors or kill signals send. So there truly must be a condition in our code that is calling cleanup twice even if there was no error the first time...

**Comment by Luap99** [comment]
@edsantiago I assume the no flake retry is the only thing I need to trigger this?

I added the parent pid info to my log as it wasn't clear if netavark teardown was run by the some podman command or two different ones.

**Comment by Luap99** [comment]
> Not to trigger it, more precisely, to see it. I think the error is happening all the time, it's just that normally we have to examine each individual log to find it. Yes, setting retries to 0 should make failures visible.

Yes that is what I meant.

**Comment by Luap99** [comment]
Interesting I found the netns error also in my log for the cleanup process. So it can happen without us ever seeing as wel and without causing a test failure. 

```
stderr 07:36:29.528707 79038    79009    podman      time="2024-08-07T07:36:29-05:00" level=error msg="Cleaning up container: failed to clean up container e22e54194869285aef9d593f0cdfff9e23d5fdfb14615dbcf71aac829d225a77: cannot get namespace path unless container edb052e52aac57efd13979ce3921481aff6baa06e556199775ab0ca6b567cdd8 is running: container is stopped"
stderr 07:36:29.528744 79038    79009    podman      Error: failed to clean up container e22e54194869285aef9d593f0cdfff9e23d5fdfb14615dbcf71aac829d225a77: cannot get namespace path unless container edb052e52aac57efd13979ce3921481aff6baa06e556199775ab0ca6b567cdd8 is running: container is stopped
ainerexec   07:36:29.529410 79114    79061    podman       /usr/libexec/podman/netavark --config /run/containers/networks --rootless=false --aardvark-binary=/usr/libexec/podman/aardvark-dns teardown /run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46
cmd    07:36:29.529780 79114    79061    /usr/libexec/podman/netavark�--config�/run/containers/networks�--rootless=false�--aardvark-binary=/usr/libexec/podman/aardvark-dns�teardown�/run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46�
exit   07:36:29.617636 79114    79061    netavark     0 0
stderr 07:36:29.638209 79038    79009    podman      time="2024-08-07T07:36:29-05:00" level=error msg="IPAM error: failed to get ips for container ID edb052e52aac57efd13979ce3921481aff6baa06e556199775ab0ca6b567cdd8 on network podman-default-kube-network"
ontainer edb052e52aac57efd13979ce3921481aff6baexec   07:36:29.638942 79150    79038    podman       /usr/libexec/podman/netavark --config /run/containers/networks --rootless=false --aardvark-binary=/usr/libexec/podman/aardvark-dns teardown /run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46
cmd    07:36:29.639236 79150    79038    /usr/libexec/podman/netavark�--config�/run/containers/networks�--rootless=false�--aardvark-binary=/usr/libexec/podman/aardvark-dns�teardown�/run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46�
exit   07:36:29.642237 79150    79038    netavark     1 0
stderr 07:36:29.642647 79038    79009    podman      time="2024-08-07T07:36:29-05:00" level=error msg="IPAM error: failed to find ip for subnet 10.89.0.0/24 on network podman-default-kube-network"
775ab0ca6b567cdd8 on network podman-defaulstderr 07:36:29.642714 79038    79009    podman      time="2024-08-07T07:36:29-05:00" level=error msg="netavark: open container netns: open /run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46: IO error: No such file or directory (os error 2)"
-network"
ontainer edb052e52aac57efd13979ce3921481aff6bastderr 07:36:29.642747 79038    79009    podman      time="2024-08-07T07:36:29-05:00" level=error msg="Unable to clean up network for container edb052e52aac57efd13979ce3921481aff6baa06e556199775ab0ca6b567cdd8: \"unmounting network namespace for container edb052e52aac57efd13979ce3921481aff6baa06e556199775ab0ca6b567cdd8: failed to remove ns path: remove /run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46: no such file or directory, failed to unmount NS: at /run/netns/netns-00a1fc0e-8ec3-0941-ba07-9f4989af1e46: no such file or directory\""
exit   07:36:29.656424 79103    78908    podman       0 0
exit   07:36:29.676194 79038    79009    podman       125 0
```
https://api.cirrus-ci.com/v1/artifact/task/5220866676490240/cleanup_tracer/podman-cleanup-tracer.log
https://api.cirrus-ci.com/v1/artifact/task/5220866676490240/html/int-podman-rawhide-root-host-sqlite.log.html

**Comment by Luap99** [comment]
@edsantiago Great news I am confident that I found the root cause, a locking/container state misuse bug that caused us to add the netns back after we cleaned it up already.
Please give https://github.com/containers/podman/pull/23487/commits/d6bcc14f50deefa4c7289b2c30149c296de08e11 a try. Hopefully this is enough, I will prepare a proper PR with a better description on Monday.

**Comment by edsantiago** [comment]
One new failure seen, [f39 root](https://api.cirrus-ci.com/v1/artifact/task/6559303622983680/html/int-podman-fedora-39-root-container-boltdb.log.html#t--Podman-network-podman-network-remove--1):
```
# podman [options] network ls --quiet
net5ef1e5e5df3c173e8948e8e487fc7c21c14438e290de0a5a04e3f8b610f60b47
networkIDTest
podman
podman-default-kube-network
time="2024-08-10T17:56:31Z" level=warning msg="Error reading network config file \"/etc/containers/networks/net25120a9e54ed259eaa0b8b853f1e8cda89d0c538a55825de0e591c2e67950554.json\": EOF"

[FAILED] Unexpected warnings seen on stderr: "time=\"2024-08-10T17:56:31Z\" level=warning msg=\"Error reading network config file \\\"/etc/containers/networks/net25120a9e54ed259eaa0b8b853f1e8cda89d0c538a55825de0e591c2e67950554.json\\\": EOF\""
```
Looks more like a race in `network ls` than anything you've changed, but still, it's a new one I don't remember seeing before.

**Comment by edsantiago** [comment]
It took a LOT of runs, but here's what looks like the same issue. [rawhide rootless](https://api.cirrus-ci.com/v1/artifact/task/5670143555010560/html/int-podman-rawhide-rootless-host-sqlite.log.html#t--Podman-UserNS-support-podman-userns-container-CTR--1):
```
$ podman [options] stop --all -t 0
Error: removing container 78e1e7c4d390f06129b5a13de8ede2351eafaf3ccc76f665177b7e5ce3bc922c network: unmounting network namespace for container 78e1e7c4d390f06129b5a13de8ede2351eafaf3ccc76f665177b7e5ce3bc922c: failed to remove ns path: remove /run/user/2153/netns/netns-1fac2ba0-0d2a-10cb-7dbe-900b0f29ae60: device or resource busy
```

**Comment by Luap99** [comment]
> It took a LOT of runs, but here's what looks like the same issue. [rawhide rootless](https://api.cirrus-ci.com/v1/artifact/task/5670143555010560/html/int-podman-rawhide-rootless-host-sqlite.log.html#t--Podman-UserNS-support-podman-userns-container-CTR--1):
> 
> ```
> $ podman [options] stop --all -t 0
> Error: removing container 78e1e7c4d390f06129b5a13de8ede2351eafaf3ccc76f665177b7e5ce3bc922c network: unmounting network namespace for container 78e1e7c4d390f06129b5a13de8ede2351eafaf3ccc76f665177b7e5ce3bc922c: failed to remove ns path: remove /run/user/2153/netns/netns-1fac2ba0-0d2a-10cb-7dbe-900b0f29ae60: device or resource busy
> ```

Note the same issue (EBUSY vs ENOENT), this should be fixed by https://github.com/containers/common/pull/2112, vendor in https://github.com/containers/podman/pull/23519

**Comment by Luap99** [comment]
Also https://github.com/containers/podman/commit/d6bcc14f50deefa4c7289b2c30149c296de08e11 is such a generic issue it would not surprise me if this fixes other weird stop flakes as well.

**Comment by Luap99** [comment]
@edsantiago @mheon PTAL, I would like to get this merged as it just helped me today again for https://github.com/containers/podman/issues/24044 so I think this is useful in general.

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,4 @@ contrib/win-installer/shasums
 contrib/win-installer/*.wixobj
 contrib/win-installer/*.wixpdb
 contrib/win-installer/*.log
+podman-cleanup-tracer.log*
```
Comment (edsantiago) [inline .gitignore:60]: Why the asterisk? (nonblocking)

**Review by edsantiago — APPROVED** [review]
Qualified LGTM (I don't grok ebpf, so I've only re-reviewed the runner changes)

**Pull request merged**